src/hooks/use-scroll-percentage-analytics.ts

```diff
@@ -0,0 +1,68 @@
+import { useEffect } from 'react'
+import { canTrackAnalytics } from 'lib/analytics'
+
+const SCROLL_PERCENTAGE_THRESHOLDS = [25, 50, 75, 90]
+
+// TODO: do we care about this potentially updating?
+const windowHeight = typeof window !== 'undefined' ? window.innerHeight : 0
```

Comment:
It seems safe to only calculate this once, happy to adjust though if it feels like an edge case we should handle.

Reply:
Yeah this is a good catch!  I feel like probably we should update this for accuracy.

Reply:
If this ends up being a big lift, we don't *absolutely* need to do this; because I think it's a smaller % of the data (and  it will come out in the wash).  I am just concerned that I am not right about it being a smaller %, and users are resizing their browsers often (I feel like I do this).